aspnet/signalr/overview/performance/using-signalr-performance-counters-in-an-azure-web-role.md

```diff
@@ -1,117 +1,143 @@
 
     [!code-cshtml[Main](using-signalr-performance-counters-in-an-azure-web-role/samples/sample6.cshtml)]
-- Create a new folder in the **WebRole1** project called Hubs. Right-click this folder, and select **Add...**, **SignalR Hub Class (v2)**. Name the new hub **MyHub** and click **Add**.
-- Open **MyHub.cs** and replace the contents with the following code.
+    
+21. Create a new folder in the **WebRole1** project named *Hubs*. Right-click the *Hubs* folder in **Solution Explorer**, select **Web &gt; SignalR**, and select **SignalR Hub Class (v2)**. Name the new hub *MyHub.cs* and click **Add**.
+
+    ![Adding SignalR Hub Class to the Hubs folder in the Add New Item dialog](using-signalr-performance-counters-in-an-azure-web-role/_static/image13.png)
+
+22. *MyHub.cs* will automatically open in the main window. Replace the contents with the following code, then save and close the file:
 
     [!code-csharp[Main](using-signalr-performance-counters-in-an-azure-web-role/samples/sample7.cs)]
-- **[Crank.exe](signalr-connection-density-testing-with-crank.md)** is a connection density testing tool provided with the SignalR codebase. Since Crank requires a **PersistentConnection**, we'll add one to the site to test. Add a new folder to the **WebRole1** project called **PersistentConnections**. Right-click this folder and select **Add...**, **Class**. Name the new class **MyPersistentConnection** and click **Add**.
-- Open **MyPersistentConnection.cs** and replace the contents with the following code.
+    
+23. *[Crank.exe](signalr-connection-density-testing-with-crank.md)* is a connection density testing tool provided with the SignalR codebase. Since Crank requires a persistent connection, you add one to your site for use when testing. Add a new folder to the **WebRole1** project called *PersistentConnections*. Right-click this folder and select **Add &gt; Class...**. Name the new class file *MyPersistentConnections.cs* and click **Add**.
+
+24. Visual Studio will open the *MyPersistentConnections.cs* file in the main window. Replace the contents with the following code, then save and close the file:
 
     [!code-csharp[Main](using-signalr-performance-counters-in-an-azure-web-role/samples/sample8.cs)]
-- Next, we'll start the SignalR objects when OWIN starts up, using the `Startup.cs` class. Your project already contains a **Startup.cs** class, unless you changed the authentication method when creating the project, in which case you'll need to create it. Open or create **Startup.cs** and replace the contents with the following code.
+    
+25. Using the `Startup` class, the SignalR objects start when OWIN starts up. Your project already contains a `Startup` class, unless you changed the authentication method when creating the project, in which case you need to create a `Startup` class. Open or create *Startup.cs* and replace the contents with the following code:
```

Comment:
If this makes sense, 
Let's assume they didn't change their authen method - if they were smart enough to do that, that'll be smart enough to figure this out - so we don't need to call that out.

I'd be tempted to rewrite along these lines: (or something more accurate)

Open or create *Startup.cs* and replace the contents with the following code:

 [!code-csharp[Main](using-signalr-performance-counters-in-an-azure-web-role/samples/sample9.cs)]

In the code above, the `OwinStartup` attribute marks this class to start OWIN. The `Configuration` method starts SignalR.

MapSignalR - doesn't start SignalR - it maps it right? You might need to rework it.